### PR TITLE
Fixing error in jit cuda on ROCm: non-constant-expression cannot be n…

### PR DESCRIPTION
**Pull request by amathews-amd**
On ROCm, the error when compiling was "non-constant-expression cannot be narrowed from type 'int' to 'uint32_t'"
when compiling grid_reduction.cu. 

Added typecast to fix issue. 

Also, removed test skip with ROCm : re-enabling

### CODE REVIEW

**Comment by amathews-amd** [comment]
@jithunnair-amd @pruthvistony @jeffdaily @sunway13 - please review

**Review by jithunnair-amd — APPROVED** [review]
LGTM

**Comment by ngimel** [comment]
@csarofeen for changes in grid_reduction.cu

**Comment by amathews-amd** [comment]
@ngimel @csarofeen @malfet Would appreciate if this is reviewed soon.
Thanks, Aswin.

**Review by csarofeen — APPROVED** [review]
Seems fine to me.

File: torch/csrc/jit/codegen/cuda/runtime/grid_reduction.cu [hunk 1]
```diff
@@ -64,9 +64,9 @@ offset(const _dim3pos& pos, const _dim3dim& dim) {
 template <bool X_BLOCK, bool Y_BLOCK, bool Z_BLOCK, typename _dim3>
 __device__ dim3 dimension_of_reduction_segment(const _dim3& grid_dim) {
   return dim3{
-      X_BLOCK ? grid_dim.x : 1,
-      Y_BLOCK ? grid_dim.y : 1,
-      Z_BLOCK ? grid_dim.z : 1};
+      X_BLOCK ? (unsigned)grid_dim.x : 1,
+      Y_BLOCK ? (unsigned)grid_dim.y : 1,
+      Z_BLOCK ? (unsigned)grid_dim.z : 1};
```
Comment (malfet) [inline torch/csrc/jit/codegen/cuda/runtime/grid_reduction.cu:69]: Don't we need to make sure that 1 is unsigned constant, aren't we?
```suggestion
      X_BLOCK ? (unsigned)grid_dim.x : 1U,
      Y_BLOCK ? (unsigned)grid_dim.y : 1U,
      Z_BLOCK ? (unsigned)grid_dim.z : 1U};
```

Comment (ngimel) [reply torch/csrc/jit/codegen/cuda/runtime/grid_reduction.cu]: `dim3` constructor will turn it into unsigned.

File: torch/csrc/jit/codegen/cuda/runtime/grid_reduction.cu [hunk 2]
```diff
@@ -125,9 +125,9 @@ offset_in_reduction_segment(const _dim3bi& block_idx, const _dim3gd& grid_dim) {
 template <bool X_THREAD, bool Y_THREAD, bool Z_THREAD, typename _dim3>
 __device__ dim3 dimension_of_reduction_block(const _dim3& block_dim) {
   return dim3{
-      X_THREAD ? block_dim.x : 1,
-      Y_THREAD ? block_dim.y : 1,
-      Z_THREAD ? block_dim.z : 1};
+      X_THREAD ? (unsigned)block_dim.x : 1,
+      Y_THREAD ? (unsigned)block_dim.y : 1,
+      Z_THREAD ? (unsigned)block_dim.z : 1};
```
Comment (ngimel) [inline torch/csrc/jit/codegen/cuda/runtime/grid_reduction.cu:130]:
```suggestion
      X_THREAD ? (unsigned)block_dim.x : 1U,
      Y_THREAD ? (unsigned)block_dim.y : 1U,
      Z_THREAD ? (unsigned)block_dim.z : 1U};
```

**Pull request closed without merging**